sdcm/cluster_aws.py

```diff
@@ -367,52 +367,95 @@ def network_config_ipv6_workaround_script():  # pylint: disable=invalid-name
 
     @staticmethod
     def configure_eth1_script():
-        return dedent("""
-            BASE_EC2_NETWORK_URL=http://169.254.169.254/latest/meta-data/network/interfaces/macs/
-            NUMBER_OF_ENI=`curl -s ${BASE_EC2_NETWORK_URL} | wc -w`
-            for mac in `curl -s ${BASE_EC2_NETWORK_URL}`
-            do
-                DEVICE_NUMBER=`curl -s ${BASE_EC2_NETWORK_URL}${mac}/device-number`
-                if [[ "$DEVICE_NUMBER" == "1" ]]; then
-                   ETH1_MAC=${mac}
+        return dedent(r"""
+            if `grep -qi "ubuntu" /etc/os-release`; then
+
+                ETH1_IP_ADDRESS=`ip route show | grep eth1 | grep -oPm1 'src \K[0-9]*\.[0-9]*\.[0-9]*\.[0-9]*'`
+                ETH1_CIDR_BLOCK=`ip route show | grep eth1 | grep -oPm1 '\K[0-9]*\.[0-9]*\.[0-9]*\.[0-9]*/[0-9]*'`
+                ETH1_SUBNET=`echo ${ETH1_CIDR_BLOCK} | grep -oP '\\K/\\d+'`
```

Comment:
Does it need to support ipv6?

Reply:
In theory it should, but we can do it in a followup PR.